### ISSUE TITLE
Check rust lints when an unknown lint is detected

### ISSUE THREAD

**Issue by alice-i-cecile** [rust-lang/rust#118183]
### Code

```toml
[lints.clippy]
missing_docs = "warn"
```

In your Cargo.toml.

### Current output

```Shell
warning[E0602]: unknown lint: `clippy::missing_docs`
  |
  = help: did you mean: `clippy::erasing_op`
  = note: requested on the command line with `-W clippy::missing_docs`
  = note: `#[warn(unknown_lints)]` on by default
```

### Desired output

```Shell
warning[E0602]: unknown lint: `clippy::missing_docs`
  |
  = help: did you mean: `missing_docs`. A lint with the same name was found in `rustc` lints.
  = note: requested on the command line with `-W clippy::missing_docs`
  = note: `#[warn(unknown_lints)]` on by default
```

### Rationale and extra context

Initially opened at https://github.com/rust-lang/rust-clippy/issues/11855
